src/plugins/plugin.colorschemes.js

```diff
@@ -20,16 +20,32 @@ export default {
 	beforeUpdate: function(chart, options) {
 		var s = options.scheme.split('.');
 		var category = colorschemes[s[0]];
-		var scheme, length, colorIndex, color;
+		var scheme, schemeClone, length, colorIndex, color;
 
 		if (category) {
 			scheme = category[s[1]];
-			length = scheme.length;
+
 			if (scheme) {
+
+				// clone the original scheme
+				schemeClone = scheme.slice();
+
+				if (options.custom) {
+					// Execute own custom color function
+					var colorFunctionResult = options.custom(schemeClone);
```

Comment:
The utility function `helpers.callback` is available in Chart.js, so let's use it. This function checks if `options.custom ` is defined and it is a function, and returns `undefined` if not. So, we don't need `if (options.custom)` in the previous line. 
```js
var colorFunctionResult = helpers.callback(options.custom, [schemeClone]);
```

Reply:
Great, didn't know that utility function. I added it.